robot_log_visualizer/robot_visualizer/meshcat_provider.py

```diff
@@ -138,34 +139,71 @@ def find_model_joints(model_name, considered_joints):
 
                     if model_filenames:
                         model_found_in_env_folders = True
-                        self.custom_model_path = str(model_filenames[0])
+                        self.model_path = str(model_filenames[0])
                         break
 
             # If the model is not found we exit
             if not model_found_in_env_folders:
                 return False
 
             self.model_joints_index = find_model_joints(
-                self.custom_model_path, considered_joints
+                self.model_path, considered_joints
             )
             considered_model_joints = [
                 considered_joints[i] for i in self.model_joints_index
             ]
             model_loader.loadReducedModelFromFile(
-                self.custom_model_path, considered_model_joints
+                self.model_path, considered_model_joints
             )
 
         if not model_loader.isValid():
             return False
 
+        self.meshcat_visualizer_mutex.lock()
+
+        # if the model is already loaded we remove it
+        if self._is_model_loaded:
+            self._meshcat_visualizer.delete(shape_name="robot")
+            self._is_model_loaded = False
+
+        model = model_loader.model()
+        if base_frame is None:
+            self.frame_T_base = np.eye(4)
+            self.base_frame = model.getFrameName(model.getDefaultBaseLink())
+            link_frame = None
+        else:
+            base_frame_index = model.getFrameIndex(base_frame)
+            frame_T_base = model.getFrameTransform(base_frame_index)
+            frame_T_base = frame_T_base.inverse().asHomogeneousTransform()
+            self.frame_T_base = frame_T_base.toNumPy()
+
+            link_frame = model.getFrameName(model.getFrameLink(base_frame_index))
```

Comment:
Nitpick. It may be personal taste, but I find it easier to  read if you use `getLinkName` here:
```suggestion
            link_frame = model.getLinkName(model.getFrameLink(base_frame_index))
```

but feel free to ignore.